### PR TITLE
add DBNet MobileNetv3 PP-OCRv3 model

### PR DESCRIPTION
**Pull request by tonytonglt**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/models/backbones/det_mobilenet.py [hunk 1]
```diff
@@ -59,6 +59,254 @@ def construct(self, x: Tensor) -> List[Tensor]:
...
+        if self.if_se:
+            x = self.mid_se(x)
+        x = self.linear_conv(x)
+        if self.if_shortcut:
+            x = ops.add(inputs, x)
+        return x
+
+
+class SEModule(nn.Cell):
+    def __init__(self, in_channels, reduction=4):
```
Comment (panshaowu) [inline mindocr/models/backbones/det_mobilenet.py:282]: `mindocr/models/backbones/mindcv_models/layers/squeeze_excite.py`
中`SqueezeExcite`是否可替换？

File: mindocr/models/necks/fpn.py [hunk 2]
```diff
@@ -306,3 +306,107 @@ def construct(self, features):
 
         out = self.relu7(self.bn7(self.conv7(out)))
         return out
+
+
+class SEModule(nn.Cell):
```
Comment (panshaowu) [inline mindocr/models/necks/fpn.py:311]: 与前文的`SEModule`代码重复，建议抽取为公共类，放置在类似`common`的文件夹中。

File: tools/infer/text/preprocess.py [hunk 3]
```diff
@@ -50,10 +50,28 @@ def __init__(self, task="det", algo="DB", **kwargs):
                 },
                 {"ToCHWImage": None},
             ]
-            _logger.info(f"Pick optimal preprocess hyper-params for det algo {algo}:\n {pipeline[1]}")
-            # TODO: modify the base pipeline for non-DBNet network if needed
+
             # if algo == 'DB++':
```
Comment (panshaowu) [inline tools/infer/text/preprocess.py:54]: 此处注释代码如无明确目的，则删除。

File: tools/infer/text/preprocess.py [hunk 4]
```diff
@@ -50,10 +50,28 @@ def __init__(self, task="det", algo="DB", **kwargs):
...
             #    pipeline[1]['DetResize']['limit_side_len'] = 1152
+
+            if algo == "DB_PPOCRv3":
+                pipeline = [
+                    {"DecodeImage": {"img_mode": "RGB", "keep_ori": True, "to_float32": False}},
```
Comment (panshaowu) [inline tools/infer/text/preprocess.py:59]: 字典形式与后文的`NormalizeImage`调整至统一。

File: mindocr/models/backbones/det_mobilenet.py [hunk 5]
```diff
@@ -59,6 +59,254 @@ def construct(self, x: Tensor) -> List[Tensor]:
...
+                [5, 576, 96, True, 'hardswish', 1],
+                [5, 576, 96, True, 'hardswish', 1],
+            ]
+            cls_ch_squeeze = 576
+        else:
+            raise NotImplementedError("mode[" + arch +
+                                      "_model] is not implemented!")
+
+        supported_scale = [0.35, 0.5, 0.75, 1.0, 1.25]
+        assert scale in supported_scale, \
```
Comment (panshaowu) [inline mindocr/models/backbones/det_mobilenet.py:128]: 尽量避免在可能用于生产环境的代码中，使用assert。

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+5. 通过可微分二值化将概率图和阈值图合并为一个近似二值图单元近似二值图，并生成文本边界框。
+
+该DBNet-PPOCRv3网络参考自PP-OCRv3 [<a href="#参考文献">2</a>] 的检测模块。其中针对与DBNet的优化主要有：
+ - LK-PAN：大感受野的PAN结构；
+ - DML：教师模型互学习策略；
+ - RSE-FPN：残差注意力机制的FPN结构；
+
+## 2. 权重转换
+
+如您已经有训练好的模型，想在MindOCR下直接进行推理或进行微调续训，您可以对训练好的模型转换为MindSpore格式的ckpt文件。
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+
+运行convert_helper.py脚本，导入需要进行转换的pdparams文件、权重名字对应关系json文件和ckpt输出路径，即可进行权重转换。
+
+```shell
+python tools/convert_helper.py \
+-i path/to/paddleocr.pdparams \
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+          is_hwc: True
+          mean: imagenet
+          std: imagenet
+      - ToCHWImage:
+  ...
+```
+
+- `DecodeImage` 和 `DetLabelEncode`：这两个转换函数解析 `train_det.txt` 文件中的字符串，加载图像和标签，并将它们保存为一个字典；
+
+- `RandomColorAdjust`， `RandomHorizontalFlip`， `RandomRotate`， `RandomScale` 和 `RandomCropWithBBox`：这些转换函数执行典型的图像增强操作。除了 `RandomColorAdjust`以外，其他函数都会改变边界框标签。
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md:178]: 这几个是通用的data_transform的pipeline，在函数内部有功能说明

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+
+- `ValidatePolygons`：它过滤掉由于之前的数据增强而在出现图像外部的边界框；
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+  neck:
+    name: RSEFPN
+    out_channels: 96
+    shortcut: True
+  head:
+    name: DBHeadEnhance
+    k: 50
+    bias: False
+    adaptive: True
+  pretrained: /local_path_to_the_ckpt/dbnet_mobilenetv3_ppocrv3-70d6018f.ckpt
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+
+然后运行：
+
+```shell
+python tools/eval.py --config configs/det/dbnet/db_mobilenetv3_ppocrv3.yaml
+```
+
+### 3.4 模型推理
+
+用户可以通过使用推理脚本快速得到模型的推理结果。请先将图片放至在同一文件夹内，然后执行
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/README_CN_PP-OCRv3.md [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+
+```yaml
+system:
+  distribute: False # During evaluation stage, set to False
+...
+eval:
+  ckpt_load_path: /local_path_to_the_ckpt/model.ckpt
+```
+
+然后运行：
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/README_CN_PP-OCRv3.md]: 已修改

File: configs/det/dbnet/db_mobilenetv3_ppocrv3.yaml [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+          thresh_max: 0.7
+      - NormalizeImage:
+          bgr_to_rgb: False
+          is_hwc: True
+          mean: imagenet
+          std: imagenet
+      - ToCHWImage:
+    #  the order of the dataloader list, matching the network input and the input labels for the loss function, and optional data for debug/visualize
+    output_columns: [ 'image', 'binary_map', 'mask', 'thresh_map', 'thresh_mask']
+#    output_columns: ['image'] # for debug op performance
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/db_mobilenetv3_ppocrv3.yaml]: 已修改

File: configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json [hunk 14]
```diff
@@ -0,0 +1,303 @@
+{
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json:1]: 已在文档中添加说明

File: mindocr/models/backbones/det_mobilenet.py [hunk 15]
```diff
@@ -59,6 +59,254 @@ def construct(self, x: Tensor) -> List[Tensor]:
...
+    def construct(self, x):
+        x = self.conv(x)
+        out_list = []
+        for stage in self.stages:
+            x = stage(x)
+            out_list.append(x)
+        return out_list
+
+
+class ConvBNLayer(nn.Cell):
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply mindocr/models/backbones/det_mobilenet.py:188]: 功能不一致

File: mindocr/models/backbones/det_mobilenet.py (hunk 1 above)
Comment (tonytonglt) [reply mindocr/models/backbones/det_mobilenet.py]: 功能不一致

File: mindocr/models/necks/fpn.py (hunk 2 above)
Comment (tonytonglt) [reply mindocr/models/necks/fpn.py]: 已提取为公共模块

File: mindocr/models/backbones/det_mobilenet.py (hunk 5 above)
Comment (tonytonglt) [reply mindocr/models/backbones/det_mobilenet.py]: 已修改为raise ValueError

File: configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json
Comment (SamitHuang) [inline configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json:1]: for model conversion, suggest change from _helper.json -> _param_map.json

**Review by SamitHuang — APPROVED** [review]

File: configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json
Comment (tonytonglt) [reply configs/det/dbnet/db_mobilenetv3_ppocrv3_helper.json:1]: Got it. Modified.

**Review by VictorHe-1 — APPROVED** [review]

**Pull request merged**